service/service.proto

```diff
@@ -15,32 +15,42 @@ message Service {
 }
 
 // Events are emitted by the service whenever the service wants.
+// TODO(ilgooz) remove key field when Event type crafted manually.
 message Event {
-  string name = 1;                  // The event's name.
-  string description = 2;           // The event's description.
-  map<string, Parameter> data = 3;  // The list of data of this event.
+  string key = 4;                   // Event's key.
+  string name = 1;                  // Event's name.
+  string description = 2;           // Event's description.
+  string serviceName = 5;           // Event's service name.
```

Comment:
Do you plan to keep `serviceName ` in `Event` in the future?
If not, how do you plan to access this data in the future from the Event type?

Reply:
no i don't. i'll add private service field to _Event_ type.
```go
type Event struct {
   service *Service
}
```

Reply:
I really don't like to have the service in the event, we should try to keep things really separated and ask ourselves why we really need the service in the event. If it's just to have the name of the service in one error this can be resolved with a event error that is catch and where the calling function can add the service.
Have a double dependencies between resources always makes things way more complicated in the end